remote-api/src/main/java/org/alfresco/rest/api/impl/CategoriesImpl.java

```diff
@@ -106,6 +102,37 @@ public List<Category> createSubcategories(String parentCategoryId, List<Category
                 .collect(Collectors.toList());
     }
 
+    @Override
+    public CollectionWithPagingInfo<Category> getCategoryChildren(String parentCategoryId, Parameters params)
+    {
+        final NodeRef parentNodeRef = getParentNodeRef(parentCategoryId);
+        final List<ChildAssociationRef> childCategoriesAssocs = nodeService.getChildAssocs(parentNodeRef).stream()
+                        .filter(ca -> ContentModel.ASSOC_SUBCATEGORIES.equals(ca.getTypeQName()))
+                        .collect(Collectors.toList());
+        final List<Category> categories = childCategoriesAssocs.stream()
+                .map(c -> mapToCategory(c.getChildRef()))
+                .collect(Collectors.toList());
```

Comment:
I think most of this logic should go (or already be present) in the category service. I definitely think that using the node service is the right approach though.

Reply:
I have mixed feelings about usage of `CategoryService` interface here. 
To me it looks it is mentioned for search purposes (it's even placed in `org.alfresco.service.cmr.search` package).

I'd be tempted to create a middle-tier category related service (don't have a good name for it, maybe `CategoryNodeService`) which could take most of the logic which is now put in `CategoriesImpl` but would restrain itself to DB node operations. 
If we'd need 'old' methods or search related queries, we could always call the 'old' `CategoryService` (although it already is a mix of search and DB nodes stuff).
We definitely need to discuss that soon.

Reply:
Yes - that sounds like a good solution.